### PR TITLE
fix: SpotlightPopoverTour bugs

### PR DESCRIPTION
**Pull request by anuraghazra**
Fixes 3 bugs: 

- Safari body-scroll-lock causing the page to get clipped because storybook doesn't set width/height on `body` - fixed by setting width/height
- Initial delay of opening the mask - fixed by immediately updating the mask size on initial render
- Delay of transitioning between steps which occurs because we need to wait for the animation to finish before scrolling otherwise the scroll gets interrupted - fixed by reduced this to 100ms

### CODE REVIEW

**Comment by kamleshchandnani** [comment]
@anuraghazra when i close the tour say at 2/3 then it shall restart right from 1/3? Right now on storybook it resumes from 2/3

**Comment by kamleshchandnani** [comment]
also, when I click on a story and then click stories tab for the first time on iOS it shows blank after refreshing the story appears

image

**Comment by anuraghazra** [comment]
> @anuraghazra when i close the tour say at 2/3 then it shall restart right from 1/3? Right now on storybook it resumes from 2/3

Tour is a controlled component, so this behaviour is entirely upon the product usecase. We just give them the ability to change the `activeStep`, if consumer wants the can reset the activeStep when `onFinish` callback is fired.

**Comment by anuraghazra** [comment]
> also, when I click on a story and then click stories tab for the first time on iOS it shows blank after refreshing the story appears

Not able to reproduce this with iOS simulator can you provide a video?

**Review by saurabhdaware — COMMENTED** [review]
Looks good to me. Added the exit animation delay thing in slack thread

File: packages/blade/docs/guides/Usage.stories.mdx [hunk 1]
```diff
@@ -45,6 +45,20 @@ function AppWrapper(): JSX.Element {
 export default AppWrapper;
 ```
 
+## iOS Safari Specific Setup
+
+When using BottomSheet or SpotlightPopoverTour, 
+Make sure to set a width/height to the `body` otherwise when they open, the page will get clipped. 
+
+This happens due to a bug in iOS safari where it won't compute the height of the body correctly.
```
[the remark this answers is not in the document]
Comment (anuraghazra) [reply packages/blade/docs/guides/Usage.stories.mdx:53]: I observed this weird bug, but don't have exact webkit bug reference. 

There are multiple issues with safari's height calculations: 

https://developer.apple.com/forums/thread/128949
https://stackoverflow.com/questions/68705153/height-100-stopped-working-when-run-in-safari
https://github.com/chakra-ui/chakra-ui/issues/6027

**Pull request merged**